amp/WEB-INF/src/org/digijava/kernel/ampapi/endpoints/activity/ActivityErrors.java

```diff
@@ -48,5 +48,8 @@
     public static final ApiErrorMessage FUNDING_PLEDGE_ORG_GROUP_MISMATCH = new ApiErrorMessage(30, 
             "The organization group of the pledge doesn't match with funding donor organization group");
     public static final ApiErrorMessage FIELD_TOO_MANY_VALUES_NOT_ALLOWED = new ApiErrorMessage(31, "Too many values");
+    public static final ApiErrorMessage ACTIVITY_NOT_FOUND = new ApiErrorMessage(32,
+            "Activity not found");
```

Comment:
Please rather use `AuthRule.VIEW_ACTIVITY` rule to report security or not found use case.

Reply:
the endpoint needs to be used for a non logged in user (coming from the portal, or directly from the link) so thats why I didn't use the VIEW_ACTIVITY AuthRule

Reply:
Sounds good then, 404 would clarify. Please note however that VIEW_ACTIVITY does multiple things that now you need to do else where:
1. Activity exists (that you did)
2. Activity is the latest version. Where will the redirect to the latest be handled? E.g. an AMP filter can do that or in info EP response we can provide the latest and UI will redirect.
3. I can see currently private activities can be seen by public users (e.g. shared linked from someone else or logged out). If it is a bug, then we will need a rule to check for private activities as well.

Reply:
for 2 I will add extra info saying its not the last version since currently the preview(old) if you see a version that is not the last, it says *Requested activity version was not the latest version. Preview switched to showing the last version!* you refresh and it never shows the last version. I will ask dani slightly modify the text and add link so a the new version is called.

Regarding private ws you are right, it does not do  the check, but in the current preview you are able to see the private activity if you are not logged in. You get the message but you see it anyways. Ill check with Vanessa if we should fix now , or as an improvement.